File: Programmers/Javascript/level1/소수찾기.js
function solution(n) {
    if(n==2){
        return 1
    }

    var count=1
    var s=false
    for(var i = 3; i<=n;i++)
        {
            for(var j=2; j<=i-1; j++)
                {
                    if(i%j==0)
                        {
                           s=false
                           break;
                        }
                    s=true
                }
            if(s){
                count+=1

            }
        }
    return count;
}

File: Programmers/Python3/lv3/소수 찾기.py
from itertools import permutations
numbers="55"

def creat_p(numbers):
    li=set()
    for i in range(1,len(numbers)+1):
        for j in permutations(numbers,i):
            li.add(int(''.join(j)))
    return list(li)

def solution(numbers):
    answer=0
    p=creat_p(numbers)
    print("p:",p)
    max_p=max(p)
    primes=[]
    is_prime=[True]*(max_p+1)
    is_prime[0]=False
    is_prime[1]=False
    for i in range(2,max_p):
        if is_prime[i]:
            primes.append(i)
            for j in range(i*i,max_p+1,i):
                is_prime[j] = False
    
    for i in p:
        if is_prime[int(i)]:
            answer+=1
        
    return answer

print(solution(numbers))